systemcontractindex/stakingindex/index.go

```diff
@@ -48,15 +48,21 @@ type (
 		cache         *cache // in-memory cache, used to query index data
 		mutex         sync.RWMutex
 		blockInterval time.Duration
+		bucketNS      string
+		ns            string
 	}
 )
 
 // NewIndexer creates a new staking indexer
 func NewIndexer(kvstore db.KVStore, contractAddr string, startHeight uint64, blockInterval time.Duration) *Indexer {
+	bucketNS := stakingBucketNS + "#" + contractAddr
+	ns := stakingNS + "#" + contractAddr
 	return &Indexer{
-		common:        systemcontractindex.NewIndexerCommon(kvstore, stakingNS, stakingHeightKey, contractAddr, startHeight),
+		common:        systemcontractindex.NewIndexerCommon(kvstore, ns, stakingHeightKey, contractAddr, startHeight),
```

Comment:
move these 2 `bucketNS, ns` into `newCache()`, so don't need to store them in `Indexer`, and L76 can keep the same